### PR TITLE
Changes for different petfriends pricelist

### PR DESCRIPTION
**Pull request by pascalckunz**
add methods to filter pricelists with timestamps
delphin uploads 2 files with a timestamp each day.
these files have to be taken from the specific directory
the old files have to be deleted
the functions for s3 and url download are not neccessary

### CODE REVIEW

File: src/app/code/community/Ambimax/PriceImport/Helper/Data.php [hunk 1]
```diff
@@ -117,11 +118,172 @@ public function getErpImporter(bool $load = false): Ambimax_PriceImport_Model_Er
...
-                $this->_import->loadCsvData(
-                    Mage::getStoreConfig('ambimax_priceimport/erp_import_options/file_location')
-                );
+                $this->_import->loadCsvData();
             }
         }
         return $this->_import;
     }
+
+    public function getCurrentFile(): array
```
Comment (Wysselbie) [inline src/app/code/community/Ambimax/PriceImport/Helper/Data.php]: Why is it in singular? You return an array of file names here.

File: src/app/code/community/Ambimax/PriceImport/Helper/Data.php [hunk 2]
```diff
@@ -117,11 +118,172 @@ public function getErpImporter(bool $load = false): Ambimax_PriceImport_Model_Er
...
+
+    }
+
+    /**
+     * @param $originalFilenames
+     */
+    public function prepareFileNames($originalFilenames)
+    {
+        if (empty($originalFilenames)) {
+            return;
```
Comment (Wysselbie) [inline src/app/code/community/Ambimax/PriceImport/Helper/Data.php]:
```suggestion
            return null;
```
Do not return nothing.
This prevents an error like "try to use return value of void function".

File: src/app/code/community/Ambimax/PriceImport/Helper/Data.php [hunk 3]
```diff
@@ -117,11 +118,172 @@ public function getErpImporter(bool $load = false): Ambimax_PriceImport_Model_Er
...
+    {
+
+        return Mage::getStoreConfig('ambimax_priceimport/erp_import_options/file_sftp_name');
+
+    }
+
+    /**
+     * @param $originalFilenames
+     */
+    public function prepareFileNames($originalFilenames)
```
Comment (Wysselbie) [inline src/app/code/community/Ambimax/PriceImport/Helper/Data.php]:
```suggestion
    public function prepareFileNames($originalFilenames): ?array
```

File: src/app/code/community/Ambimax/PriceImport/Helper/Data.php [hunk 4]
```diff
@@ -117,11 +118,172 @@ public function getErpImporter(bool $load = false): Ambimax_PriceImport_Model_Er
...
+    {
+        if (empty($originalFilenames)) {
+            return;
+        }
+        if (!empty($originalFilenames)) {
+            foreach ($originalFilenames as $name) {
+                $filenames[$name] = substr($name, strripos($name, '/') + 1);
+            }
+        }
```
Comment (Wysselbie) [inline src/app/code/community/Ambimax/PriceImport/Helper/Data.php]:
```suggestion
        foreach ($originalFilenames as $name) {
            $filenames[$name] = substr($name, strripos($name, '/') + 1);
        }
```
It is never empty after the first `if`.

File: src/app/code/community/Ambimax/PriceImport/Helper/Data.php [hunk 5]
```diff
@@ -117,11 +118,172 @@ public function getErpImporter(bool $load = false): Ambimax_PriceImport_Model_Er
...
+        $onlyPetfriendsFileName = $this->cleanFileNames($allFileNames, 1);
+        $fileNames = [];
+        array_push($fileNames, $fileName, $onlyPetfriendsFileName);
+        return $fileNames;
+    }
+
+    /*
+     * use $hasSuffix = 1 for list with suffix
+     */
+    public function cleanFileNames(array $fileNames, int $hasSuffix = 0): string
```
Comment (Wysselbie) [inline src/app/code/community/Ambimax/PriceImport/Helper/Data.php]:
```suggestion
    public function cleanFileNames(array $fileNames, bool $hasSuffix = false): string
```
Why not bool?

File: src/app/code/community/Ambimax/PriceImport/Helper/Data.php [hunk 6]
```diff
@@ -117,11 +118,172 @@ public function getErpImporter(bool $load = false): Ambimax_PriceImport_Model_Er
...
+    {
+        $fileName = $this->cleanFileNames($allFileNames);
+        $onlyPetfriendsFileName = $this->cleanFileNames($allFileNames, 1);
+        $fileNames = [];
+        array_push($fileNames, $fileName, $onlyPetfriendsFileName);
+        return $fileNames;
+    }
+
+    /*
+     * use $hasSuffix = 1 for list with suffix
```
Comment (Wysselbie) [inline src/app/code/community/Ambimax/PriceImport/Helper/Data.php]:
```suggestion
     * use $hasSuffix = true for list with suffix
```

File: src/app/code/community/Ambimax/PriceImport/Helper/Data.php [hunk 7]
```diff
@@ -117,11 +118,172 @@ public function getErpImporter(bool $load = false): Ambimax_PriceImport_Model_Er
...
+        if (!empty($fileNames)) {
+            $this->cleanFtpFiles($fileNames);
+        }
+        return $newestFileName;
+    }
+
+    public function cleanFtpFiles(array $fileNames): void
+    {
+        foreach ($fileNames as $filename) {
+            $this->deleteFtpFiles($this->getFtpFilesLocation() . $filename);
```
Comment (Wysselbie) [inline src/app/code/community/Ambimax/PriceImport/Helper/Data.php]: What if there is no trailing `/` in `'ambimax_priceimport/erp_import_options/file_sftp_path'`?

File: src/app/code/community/Ambimax/PriceImport/Helper/Data.php [hunk 8]
```diff
@@ -117,11 +118,172 @@ public function getErpImporter(bool $load = false): Ambimax_PriceImport_Model_Er
...
+        }
+    }
+
+    public function deleteFtpFiles($filePath): void
```
Comment (Wysselbie) [inline src/app/code/community/Ambimax/PriceImport/Helper/Data.php]:
```suggestion
    public function deleteFtpFile($filePath): void
```

File: src/app/code/community/Ambimax/PriceImport/Helper/Data.php [hunk 9]
```diff
@@ -117,11 +118,172 @@ public function getErpImporter(bool $load = false): Ambimax_PriceImport_Model_Er
...
+    }
+
+    protected function getFtpFilesLocation(): string
+    {
+
+        return Mage::getStoreConfig('ambimax_priceimport/erp_import_options/file_sftp_path');
+
+    }
+
+    protected function getFtpFileNames(): string
```
Comment (Wysselbie) [inline src/app/code/community/Ambimax/PriceImport/Helper/Data.php]: Is there a way, where the return value has more than one file name? Use singular in the whole file for this name, if not.

File: src/app/code/community/Ambimax/PriceImport/Helper/Data.php [hunk 10]
```diff
@@ -117,11 +118,172 @@ public function getErpImporter(bool $load = false): Ambimax_PriceImport_Model_Er
...
+        //@codingStandardsIgnoreEnd
+    }
+
+    public function isPriceFile(string $fileName): bool
+    {
+        $fileNameSetting = $this->getFtpFileNames();
+        if (!str_starts_with($fileName, $fileNameSetting)) {
+            return false;
+        }
+        if (!str_starts_with($fileName, $fileNameSetting . $this->_listsuffix)) {
```
Comment (Wysselbie) [inline src/app/code/community/Ambimax/PriceImport/Helper/Data.php:272]: Is it really something like `price-list-file-name.csv-fwpf-`? Seems weird to me.

File: src/app/code/community/Ambimax/PriceImport/Model/ErpImport.php [hunk 11]
```diff
@@ -64,42 +64,41 @@ public function setPriceData($data, $doMapping = true)
...
+            $io = $this->getCsvStream($fileLocation, $fileName);
 
-            //build row array
-            $row = array_combine($columns, $csvLine);
-            if (!$sku = $row['sku']) {
-                continue;
-            }
+            $data = array();
+            $columns = null;
+            $map = array_flip($this->_map);
```
Comment (Wysselbie) [inline src/app/code/community/Ambimax/PriceImport/Model/ErpImport.php]: Is it right, that `$data` will be reset every iteration? You return it with this method. Here only the data of the last iteration would be returned.
And what about multiple entries with the same SKU in different files?
And `$map` does not have to be in the `foreach`, too, in my opinion.

File: src/app/code/community/Ambimax/PriceImport/Model/ErpImport.php [hunk 12]
```diff
@@ -109,9 +108,9 @@ public function loadCsvData($fileLocation)
      * @param $additional boolean
      * @return string
      */
-    public function getLocalFilePath()
+    public function getLocalFilePath($fileName)
     {
-        return Mage::getStoreConfig('ambimax_priceimport/erp_import_options/file_path');
+        return Mage::getStoreConfig('ambimax_priceimport/erp_import_options/file_path') . $fileName;
```
Comment (Wysselbie) [inline src/app/code/community/Ambimax/PriceImport/Model/ErpImport.php:110]: What about a missing trailing `/` in `'ambimax_priceimport/erp_import_options/file_path'`?

File: src/app/code/community/Ambimax/PriceImport/Model/ErpImport.php [hunk 13]
```diff
@@ -148,7 +147,7 @@ protected function _downloadSftpFile()
         $io = new Varien_Io_File();
         $io->checkAndCreateFolder(dirname($destination));
 
-        $connectionString = str_replace(array_keys($options), $options, 'sftp://{username}:{password}@{host}{path}');
+        $connectionString = str_replace(array_keys($options), $options, 'sftp://{username}:{password}@{host}{path}' . $fileName);
```
Comment (Wysselbie) [inline src/app/code/community/Ambimax/PriceImport/Model/ErpImport.php:147]: What about a missing trailing `/` in `{path}`?

**Review by Wysselbie — CHANGES_REQUESTED** [review]

File: src/app/code/community/Ambimax/PriceImport/Helper/Data.php (hunk 7 above)
Comment (pascalckunz) [reply src/app/code/community/Ambimax/PriceImport/Helper/Data.php]: then it does not work

File: src/app/code/community/Ambimax/PriceImport/Helper/Data.php (hunk 10 above)
Comment (pascalckunz) [reply src/app/code/community/Ambimax/PriceImport/Helper/Data.php:272]: its artikelstamm-fwpf-20210622010002

Comment (pascalckunz) [reply src/app/code/community/Ambimax/PriceImport/Helper/Data.php:272]: you only use the filename start in the administration fields

Comment (Wysselbie) [reply src/app/code/community/Ambimax/PriceImport/Helper/Data.php:272]: Aaaahhh

**Review by Wysselbie — APPROVED** [review]

**Pull request merged**